core/src/main/java/me/lokka30/treasury/plugin/core/utils/PluginVersion.java

```diff
@@ -0,0 +1,151 @@
+package me.lokka30.treasury.plugin.core.utils;
+
+import java.util.Objects;
+import me.lokka30.treasury.plugin.core.logging.Logger;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A version object, representing Treasury plugin's version.
+ *
+ * @author MrIvanPlays
+ */
+public final class PluginVersion {
+
+    private final short majorRevision, minorRevision, patchRevision;
+    private final boolean developmentVersion;
+    private final String stringVersion;
+
+    public PluginVersion(@NotNull String input, @NotNull Logger logger) {
+        Objects.requireNonNull(input, "input");
+        Objects.requireNonNull(logger, "logger");
+        this.stringVersion = input;
+        if (input.endsWith("-SNAPSHOT")) {
+            developmentVersion = true;
+            input = input.replace("-SNAPSHOT", "");
+        } else {
+            developmentVersion = false;
+        }
+        if (input.indexOf('.') == -1) {
+            throw new IllegalArgumentException(
+                    "Illegal Treasury version found! " +
+                            "If this is a live environment version or you're a developer doing changes, please fix this ASAP!"
+            );
+        }
+        String[] parts = input.split("\\.");
+        if (parts.length != 3) {
+            throw new IllegalArgumentException(
+                    "Illegal Treasury version found! " +
```

Comment:
Since this String is used twice, should we store it somewhere?

Reply:
no need.